### PR TITLE
#22250 Moving unique per site to field variable

### LINKED ISSUE #22250: Move Unique per site to a field variable
In order to give more control over unique fields, we added a config property called `uniquePerSite` that controls whether fields marked unique are unique globally or per site.

We should have created a `uniquePerSite` field variable that controls, at the field level, whether the unique field is measured on a per site basis or a global basis.  The default for this is `false` which would maintain the behavior of the unique field values in all dotCMS versions except 22.05, where the default was changed by accident.

We should revert the change to the default that was made in issue :
https://github.com/dotCMS/core/issues/21781

### PR DESCRIPTION
**Pull request by freddyucv**
After this issue: https://github.com/dotCMS/core/issues/21781

We create a config property to set if you want that the unique field was unique by site or global in dotCMS, this property was celled uniquePerSite with a default value equals to true.

Now We ara moving this config property to a Field Variable with the same name but with a default value equals to false, to keep the same behavior if the user don't do anything.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/contenttype/model/field/Field.java [hunk 1]
```diff
@@ -302,6 +302,18 @@ public Optional<FieldValueBuilder> fieldValue(final Object value) {
...
+    for (final FieldVariable fieldVariable : fieldVariables) {
+      if (fieldVariable.key().equals(fieldVariableName)) {
+        return Optional.of(fieldVariable.value());
+      }
+    }
+
+    return Optional.empty();
+  }
+
   ;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/model/field/Field.java:317]: ?

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/contenttype/model/field/Field.java (hunk 1 above)
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/model/field/Field.java:317]: What about this lonely semicolon?

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/contenttype/model/field/Field.java]: done https://github.com/dotCMS/core/pull/22304/commits/fc2a9966e4ffd8e1e5a0ebb351a0d02e13d4be9f

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Review by dsilvam — APPROVED** [review]

**Pull request merged**